### DISCUSSION TITLE
记录一次在 CentOS 7 / Redhat 7 上运行 DAE

### DISCUSSION THREAD

**Discussion by sbwml** [daeuniverse/dae#433] [Show and tell]
### 作为一个元老级的 Linux 发行版，虽然它的生命周期已经不长，但应该还有部分人在使用它。记录一次在 CentOS 7 上吃鹅的经历。

---------

## 更新内核

一、更新支持 BTF 的内核
 ```bash
 curl -s https://repo.cooluc.com/mailbox.repo > /etc/yum.repos.d/mailbox.repo
 yum makecache
 yum update kernel
 ```
*内核基于 Linux 6.1 LTS 重构（支持 BBRv2），并启用 eBPF 支持，也可自行编译，源码包：https://repo.cooluc.com/kernel/7/SRPMS/*

二、挂载 BPF
```bash
curl -s https://repo.cooluc.com/kernel/files/sys-fs-bpf.mount > /etc/systemd/system/sys-fs-bpf.mount
systemctl enable sys-fs-bpf.mount
```

三、挂载 Control Group v2
```bash
curl -s https://repo.cooluc.com/kernel/mount-cgroup2.service > /etc/systemd/system/mount-cgroup2.service
systemctl enable mount-cgroup2.service
```

四、重启系统，生效内核。

*检查内核版本，版本为 `6.1.xx-1.el7.x86_64` 则表示成功。*
```bash
uname -r
```

*如果内核版本没有变化，说明此前已经更新过内核，需要重建 grub2 引导使新内核优先级最高*

设置最新内核为默认
```bash
grub2-set-default 0
```

重建内核引导配置
```bash
grub2-mkconfig -o /boot/grub2/grub.cfg
```

---------

## 运行 DAE

五、现在可以用你知道的方式，下载 DAE 并运行使用了！

这里只是举例：

```bash
mkdir -p /opt/dae && cd /opt/dae

wget https://github.com/daeuniverse/dae/releases/download/v0.2.2/dae-linux-x86_64.zip

unzip dae-linux-x86_64.zip && rm -f dae-linux-x86_64.zip

cp example.dae config.dae

chmod 600 config.dae

DAE_LOCATION_ASSET=$(pwd) ./dae-linux-x86_64 run -c config.dae
```

当然，你需要根据  https://dae.v2raya.org/docs/current/category/configuration 去配置 `config.dae`

**Comment by dae-prow** [comment]
Thanks for opening this issue!

**Comment by stockfeng** [comment]
> ```shell
> systemctl enable sys-fs-bpf.mount
> ```

挂载bpf不成功，错误提示如下：
[root@centos-dae ~]# curl -s https://repo.cooluc.com/kernel/sys-fs-bpf.mount > /etc/systemd/system/sys-fs-bpf.mount
[root@centos-dae ~]# systemctl enable sys-fs-bpf.mount
Failed to execute operation: Cannot send after transport endpoint shutdown

**Comment by sbwml** [comment]
这个地址了   https://repo.cooluc.com/kernel/files/sys-fs-bpf.mount
